src/callinput.c

```diff
@@ -520,15 +520,30 @@ int callinput(void) {
 		break;
 	    }
 
-	    /* <Insert>, send exchange in CT mode */
+	    /* Semicolon or Alt-n (M-n), insert note in log.
+	     * In CT Compatible mode semicolon is an alias for Insert.
+	     * <Insert>, send exchange in CT mode, inactive otherwise.
+	     */
+	    case ';':
+	    case 238:
 	    case KEY_IC: {
-		if (ctcomp != 0) {
-	            /* F3 (RST macro) */
+		/* Semicolon is an alias for Insert in CT Compat mode. */
+		if (ctcomp != 0 && (x == KEY_IC || x == ';')) {
+		    /* F3 (RST macro) */
 		    send_standard_message(2);
-	            /* Set to space to move cursor to exchange field
-	             * which will trigger autofill if available.
-	             */
-	            x = ' ';
+
+		    /* Set to space to move cursor to exchange
+		     * field which will trigger autofill if available.
+		     */
+		    x = ' ';
+		}
+		/* When in ESM mode, semicolon opens the note dialog.
+		 * In all modes, Alt-N opens the note dialog.
+		 */
+		else if ((ctcomp == 0 && (x == ';' || x == 238))
+		          || ((ctcomp != 0) && (x == 238))) {
+		    include_note();
+		    x = -1;
 		}
 		break;
 	    }
```

Comment:
That makes the code much more complex and much less clear. Maintaining this woven set of conditions will become difficult.
I would recommend not to fold all that functions together into one case and then inside it split it again to different behavior. E.g. the Alt-N could stay out and have a case for itself even if that means to have two places which calls 'include_note()'. Maybe the same idea should be applied to the ';' key.

Reply:
an alternative (IMO still readable):
```
#define ALT_N  238  // to avoid "magic" numbers

   if (x == ALT_N || (ctcomp && x == ';')) {
      ....
}
```

but splitting per-keycode is surely clearer

Reply:
Fair enough, Tom.  I'll revisit this, possibly after this coming weekend's ARRL 160m 'test, but maybe sooner as time permits.

The idea of not using magic numbers should have been resolved long ago as there are many, always preceded by comments, however, but something that should be done regardless.